### PR TITLE
Updating Course Descriptions

### PR DESCRIPTION
**Pull request by jacquelinecai**
# Summary

This PR works on the updating course descriptions portion of the similar courses feature.

## PR Type

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Mobile + Desktop Screenshots & Recordings
Description Added:
image
Description Not Added:
image

## QA - Test Plan
- Search up any of the following courses in the Course database, and there should be an additional `classDescription` field at the bottom of the Course object

## Breaking Changes & Notes
- Wrote script to add course descriptions to database
- Removed invalid semesters from courses in database
- Added new endpoint to link with button on frontend admin page
Diagram:
image

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 notion
- [x] 🍕 in codebase
- [ ] 📕 ...
- [ ] 🙅 no documentation needed

## What GIF represents this PR?

[gif](https://www.icegif.com/wp-content/uploads/tom-and-jerry-icegif-4.gif)

### CODE REVIEW

File: client/src/modules/Admin/Components/Admin.tsx [hunk 1]
```diff
@@ -235,6 +237,24 @@ export const Admin = () => {
...
   }
 
+  /**
+   * Call when user selects "Update Descriptions" button. Scrapes the Course API 
+   * to retrieve course description and stores them in the Course database.
+  */
+  function updateDescriptions() {
+    console.log('Updating course descriptions')
+    setDisableInit(true)
+    axios.post('/api/admin/courseDescriptions/add', { token: token }).then((response) => {
```
Comment (qiandrewj) [inline client/src/modules/Admin/Components/Admin.tsx:247]: In the future, when the branch updating route names is merged, we could slightly update this to something like '/api/admin/course/desc' for consistency. Not sure about the example I provided but we can model it off the other ones :)

Comment (qiandrewj) [inline client/src/modules/Admin/Components/Admin.tsx:247]: In the future, when the branch updating route names is merged, we could slightly update this to something like '/api/admin/course/desc' for consistency. Not sure about the example I provided but we can model it off the other ones :)

File: client/src/modules/Admin/Components/Admin.tsx [hunk 2]
```diff
@@ -235,6 +237,24 @@ export const Admin = () => {
...
+      if (response.status === 200) {
+        console.log('Updated all course descriptions')
+        setDisableInit(false)
+        setDescAdded(true)
```
Comment (qiandrewj) [inline client/src/modules/Admin/Components/Admin.tsx:251]: I guess a question I have is - is this descAdded state necessary? Is it used to block any other updating functions from running, or does the disableInit field do that?

**Review by qiandrewj — COMMENTED** [review]
The new methods that you added to update the database look great to me! I liked the way that you made sure to check which courses we have in the database so that we could update all of those -- a very safe way to make sure nothing ends up breaking currently when the button is clicked. Based on this example, we could also look into the semester updating buttons... Amazing progress on similar courses!

**Review by leihelen — COMMENTED** [review]
Your code looks really good and works well for me! I like how you separated out the helper functions which made it very clear for me to follow as well as easy to test! Maybe a small improvement could be to just query the course data once in the addCourseDescription function to make it more efficient. For example the first query "const course = await Classes.findOne({ _id: courseId }).exec();" can be kept and then course can be reused. Overall, the functionality works great and this feature will really help provide more accurate course information to users so thank you for working so hard on it!

File: client/src/modules/Admin/Components/Admin.tsx (hunk 2 above)
Comment (jacquelinecai) [reply client/src/modules/Admin/Components/Admin.tsx]: Resolved in other PR - could simplify this down to one updating variable.

**Review by wizhaaa — APPROVED** [review]
Awesome work here Jacqueline - sorry it took so long to get to this. 

I love the way you handled the updating and disabled states here too! 

Backend code look great too -> well-structured and everything makes sense.

**Pull request merged**